### PR TITLE
[Enhancement]Support show create for sync mv

### PR DESCRIPTION
**Pull request by ucasfl**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by Astralidea — APPROVED** [review]

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -1082,7 +1082,53 @@ private void showCreateInternalCatalogTable(ShowCreateTableStmt showStmt) throws
         try {
             Table table = db.getTable(showStmt.getTable());
             if (table == null) {
-                ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_ERROR, showStmt.getTable());
+                if (showStmt.getType() != ShowCreateTableStmt.CreateTableType.MATERIALIZED_VIEW) {
+                    ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_ERROR, showStmt.getTable());
+                } else {
+                    // For Sync Materialized View, it is a mv index inside OLAP table,
+                    // so we can not get it from database.
+                    for (Table tbl : db.getTables()) {
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1090]: There are read locks that may take a long time here, but they can be optimized later.

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 2]
```diff
@@ -1082,7 +1082,53 @@ private void showCreateInternalCatalogTable(ShowCreateTableStmt showStmt) throws
...
+                                        originStmtBuilder.append(" from ").append(olapTable.getName()).append(" group by ")
+                                                .append(groupByString);
+                                        originStmtBuilder.delete(originStmtBuilder.length() - 1, originStmtBuilder.length());
+                                        rows.add(Lists.newArrayList(showStmt.getTable(), originStmtBuilder.toString(),
+                                                "utf8", "utf8_general_ci"));
+                                    } else {
+                                        rows.add(Lists.newArrayList(showStmt.getTable(), mvMeta.getOriginStmt(),
+                                                "utf8", "utf8_general_ci"));
+                                    }
+                                    resultSet = new ShowResultSet(ShowCreateTableStmt.getMaterializedViewMetaData(), rows);
```
[the remark this answers is not in the document]
Comment (ucasfl) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1110]: No, it has been checked in here that no sync mvs with same names:
https://github.com/StarRocks/starrocks/blob/3b38fbd51bb416e5077c74496952857a1c723fd9/fe/fe-core/src/main/java/com/starrocks/alter/MaterializedViewHandler.java#L429-L449

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1110]: all right.

**Review by Astralidea — DISMISSED** [review]

**Comment by ucasfl** [comment]
I have no idea why the unit test failed.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 38 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|38|00.00%|[505, 508, 509, 510, 511, 513, 514, 515, 516, 517, 518, 520, 521, 522, 523, 524, 525, 599, 600, 1090, 1091, 1095, 1096, 1097, 1098, 1099, 1100, 1101, 1102, 1103, 1104, 1106, 1107, 1110, 1111, 1113, 1115, 1116]|

**Review by Astralidea — APPROVED** [review]

**Comment by ucasfl** [comment]
@Astralidea Hi, could you find any one else to review it?

**Review by LiShuMing — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**